source/core/src/main/com/csse3200/game/components/quests/QuestManager.java

```diff
@@ -0,0 +1,206 @@
+import java.util.Objects;
+
+
+/**
+ * Manages, tracks and updates the quests within the game.
+ * Handles the storage and retrieval of quests and integrates with the event system.
+ */
+public class QuestManager extends Component {
+    /** Map to store quests. */
+    private final HashMap<String, QuestBasic> quests;
+    /** Event service to handle global events. */
+    private final EventService eventService = ServiceLocator.getEventService();
+    /** Logger for logging quest related attributes. */
+    private static final Logger logger = LoggerFactory.getLogger(QuestManager.class);
+    /** Sound effect for quest completion. */
+    private final Sound questComplete = ServiceLocator.getResourceService().getAsset("sounds/QuestComplete.wav", Sound.class);
+    /** Map of relevant quests. As of Sprint 1 the String[] should contain only one quest as only one is accessed*/
+    private final Map<String, String[]> relevantQuests;
+
+    /**Constructs questManager instance */
+    public QuestManager() {
+        this.quests = new HashMap<>();
+        this.relevantQuests = Map.of(
+                "Cow", new String[]{"2 Task Quest"}
+        );
+        testQuests();
+    }
+
+    /**Sets up the tasks for the quests and dialogues.  */
+    private void testQuests() {
+
+        //creates test tasks
+        Task stepsTask = new Task("steps", "Take your first steps", "Just start moving!", 1);
+        Task attackTask = new Task("attack", "Swing your first sword", "Just Attack!", 1);
+        Task testKangaTask = new Task("spawnKangaBoss", "He is Coming...", "RUN", 1);
+
+        //creates single task quest
+        List<Task> tasks = List.of(stepsTask);
+        QuestBasic firstStepsQuest = new QuestBasic("First Steps","Take your first steps in this world!", tasks, false,false,null,null);
+        addQuest(firstStepsQuest);
+
+        //creates 2 task quest
+        String[] test2StepTextProg1 = new String[]{"Welcome to Animal Kingdom!", "Here let me help with your quest...","Press Spacebar!"};
+        String[] test2StepTextProg2 = new String[]{"Yippeee!", "You completed your Quest!"};
+
+        Map<DialogueKey, String[]> test2TaskQuestDialogue = Map.of(
+                new DialogueKey("Cow", 1), test2StepTextProg1,
+                new DialogueKey("Cow", 2), test2StepTextProg2
+        );
+
+        String[] test2StepCompletionTriggers = new String[]{"","spawnKangaBoss"};
+        List<Task> tasks1 = List.of(stepsTask, attackTask);
+        QuestBasic twoTaskQuest = new QuestBasic("2 Task Quest", "Move then Attack for a Test Quest", tasks1, false, false, test2TaskQuestDialogue,test2StepCompletionTriggers);
+        addQuest(twoTaskQuest);
+
+        // Creates test quest that requires completion of 2 task quest
+        List<Task> tasks3 = List.of(testKangaTask,stepsTask, attackTask);
+        QuestBasic finalQuest = new QuestBasic("Final Boss","Complete quest 1 and 2 to summon the boss", tasks3, false,false,null,null);
```

Comment:
Adding a few different constructor methods to `QuestBasic` for cases like this might help improve code readability.

Reply:
Thank you. Fixing now.